Mathlib/SetTheory/Ordinal/Veblen.lean

```diff
@@ -331,4 +340,128 @@ theorem veblen_eq_veblen_iff :
  veblenWith_eq_veblenWith_iff (isNormal_opow one_lt_omega0)
 
 end veblen
+
+/-! ### Epsilon function -/
+
+/-- The epsilon function enumerates the fixed points of `ω ^ ⬝`.
+This is an abbreviation for `veblen 1`. -/
+abbrev epsilon := veblen 1
+
+@[inherit_doc] scoped notation "ε_ " => epsilon
+
+/-- `ε₀` is the first fixed point of `ω ^ ⬝`, i.e. the supremum of `ω`, `ω ^ ω`, `ω ^ ω ^ ω`, … -/
+scoped notation "ε₀" => ε_ 0
+
+theorem epsilon_eq_deriv (o : Ordinal) : ε_ o = deriv (fun a ↦ ω ^ a) o := by
+  rw [epsilon, ← succ_zero, veblen_succ, veblen_zero]
+
+theorem epsilon0_eq_nfp : ε₀ = nfp (fun a ↦ ω ^ a) 0 := by
+  rw [epsilon_eq_deriv, deriv_zero_right]
+
+theorem epsilon_succ_eq_nfp (o : Ordinal) :
+    ε_ (Order.succ o) = nfp (fun a ↦ ω ^ a) (Order.succ (ε_ o)) := by
+  rw [epsilon_eq_deriv, epsilon_eq_deriv, deriv_succ]
+
+theorem epsilon0_le_of_omega0_opow_le (h : ω ^ o ≤ o) : ε₀ ≤ o := by
+  rw [epsilon0_eq_nfp]
+  exact nfp_le_fp (fun _ _ ↦ (opow_le_opow_iff_right one_lt_omega0).2) (Ordinal.zero_le o) h
+
+@[simp]
+theorem omega0_opow_epsilon (o : Ordinal) : ω ^ ε_ o = ε_ o := by
+  rw [epsilon_eq_deriv, (isNormal_opow one_lt_omega0).deriv_fp]
+
+/-- `ε₀` is the limit of `0`, `ω ^ 0`, `ω ^ ω ^ 0`, … -/
+theorem lt_epsilon0 : o < ε₀ ↔ ∃ n : ℕ, o < (fun a ↦ ω ^ a)^[n] 0 := by
+  rw [epsilon0_eq_nfp, lt_nfp_iff]
+
+/-- `ω ^ ω ^ … ^ 0 < ε₀` -/
+theorem iterate_omega0_opow_lt_epsilon0 (n : ℕ) : (fun a ↦ ω ^ a)^[n] 0 < ε₀ := by
+  rw [epsilon0_eq_nfp]
+  apply iterate_lt_nfp (isNormal_opow one_lt_omega0).strictMono
+  simp
+
+theorem omega0_lt_epsilon (o : Ordinal) : ω < ε_ o := by
+  apply lt_of_lt_of_le _ <| (veblen_right_strictMono _).monotone (Ordinal.zero_le o)
+  simpa using iterate_omega0_opow_lt_epsilon0 2
+
+theorem natCast_lt_epsilon (n : ℕ) (o : Ordinal) : n < ε_ o :=
+  (nat_lt_omega0 n).trans <| omega0_lt_epsilon o
+
+theorem epsilon_pos (o : Ordinal) : 0 < ε_ o :=
```

Comment:
simp?

Reply:
Can `simp` see through `abbrev`? If so, this should just be a specific case of `veblen_pos`.

Reply:
Okay sure, then no problem